### ISSUE TITLE
Unsatisfied expectations breaks execution of tests

### ISSUE THREAD

**Issue by viliusl** [wix/specs2-jmock#9]
So if you have a set of tests in a single block (multiple examples) and one of tests does not satisfy expectations - was not called at all (wrong arguments does not trigger issue), then next tests are no longer executed - with a change of 2 outcomes:
 - no failures in a suite;
 - "Framework exited unexpectedly" type of output in IntelliJ;

Afaik this is issue with migration to Specs2 3.6 and I suspect that JMock.around impl needs to be different, but did not yet have time to play with it.

To reproduce run https://github.com/wix/specs2-jmock/blob/oneOf-bug/src/test/scala/com/wixpress/common/specs2/OneOfTest.scala and see that only 1 of 2 tests is being executed, whereas in 1st test
```
checking {
        oneOf(dep).func1("1") willReturn true
      }
```
is not invoked

**Comment by viliusl** [comment]
Ok, 'sbt test' works better than IntelliJ:
```
[info] OneOfTest+ just work 2
[error]
[error] not all expectations were satisfied
[error] expectations:
[error]   ! expected once, never invoked: dep.func1("1"); returns <true>
[error] what happened before this: nothing!
[error]
[error] STACKTRACE
[error]   org.jmock.api.ExpectationError.notAllSatisfied(ExpectationError.java:27)
[error]   org.jmock.Mockery.assertIsSatisfied(Mockery.java:213)
[error]   com.wixpress.common.specs2.JMock$$anonfun$around$2$$anonfun$apply$1.apply(JMock.scala:31)
[error]   com.wixpress.common.specs2.JMock$$anonfun$around$2$$anonfun$apply$1.apply(JMock.scala:31)
[error]   org.specs2.execute.ResultExecution$class.execute(ResultExecution.scala:25)
[error]   org.specs2.execute.ResultExecution$.execute(ResultExecution.scala:120)
[error]   com.wixpress.common.specs2.JMock$$anonfun$around$2.apply(JMock.scala:31)
[error]   com.wixpress.common.specs2.JMock$$anonfun$around$2.apply(JMock.scala:31)
[error]   org.specs2.execute.ResultExecution$class.execute(ResultExecution.scala:25)
[error]   org.specs2.execute.ResultExecution$.execute(ResultExecution.scala:120)
[error]   org.specs2.execute.ResultLogicalCombinators$ResultLogicalCombinator.o$lzycompute$1(ResultLogicalCombinators.scala:27)
[error]   org.specs2.execute.ResultLogicalCombinators$ResultLogicalCombinator.org$specs2$execute$ResultLogicalCombinators$ResultLogicalCombinator$$o$1(ResultLogicalCombinators.scala:27)
[error]   org.specs2.execute.ResultLogicalCombinators$ResultLogicalCombinator.combine$1(ResultLogicalCombinators.scala:31)
[error]   org.specs2.execute.ResultLogicalCombinators$ResultLogicalCombinator.and(ResultLogicalCombinators.scala:64)
[error]   com.wixpress.common.specs2.JMock$class.around(JMock.scala:31)
[error]   com.wixpress.common.specs2.OneOfTest.around(OneOfTest.scala:10)
[error]   org.specs2.specification.AroundEach$$anonfun$aroundContext$1$$anon$4.around(EachContext.scala:64)
[error]   org.specs2.specification.Around$class.apply(Context.scala:138)
[error]   org.specs2.specification.AroundEach$$anonfun$aroundContext$1$$anon$4.apply(EachContext.scala:64)
[error]   org.specs2.specification.create.ContextualFragmentFactory$$anonfun$example$3$$anonfun$apply$1.apply(FragmentFactory.scala:106)
[error]   org.specs2.specification.create.ContextualFragmentFactory$$anonfun$example$3$$anonfun$apply$1.apply(FragmentFactory.scala:106)
[error]   org.specs2.specification.core.Execution$$anonfun$execute$2$$anonfun$apply$2.apply(Execution.scala:70)
[error]   org.specs2.specification.core.Execution$$anonfun$execute$2$$anonfun$apply$2.apply(Execution.scala:70)
[error]   org.specs2.specification.core.Execution.setResult(Execution.scala:76)
[error]   org.specs2.specification.core.Execution$$anonfun$execute$2.apply(Execution.scala:70)
[error]   org.specs2.specification.core.Execution$$anonfun$execute$2.apply(Execution.scala:70)
[error]   scala.Option.fold(Option.scala:158)
[error]   org.specs2.specification.core.Execution.execute(Execution.scala:70)
[error]   org.specs2.specification.dsl.mutable.MutableFragmentBuilder$$anonfun$duplicateExecution$1$$anonfun$apply$3.apply(MutableFragmentBuilder.scala:119)
[error]   org.specs2.specification.dsl.mutable.MutableFragmentBuilder$$anonfun$duplicateExecution$1$$anonfun$apply$3.apply(MutableFragmentBuilder.scala:109)
[error]   scalaz.$bslash$div.fold(Either.scala:57)
[error]   org.specs2.specification.dsl.mutable.MutableFragmentBuilder$$anonfun$duplicateExecution$1.apply(MutableFragmentBuilder.scala:107)
[error]   org.specs2.specification.dsl.mutable.MutableFragmentBuilder$$anonfun$duplicateExecution$1.apply(MutableFragmentBuilder.scala:99)
[error]   org.specs2.specification.core.Execution$$anonfun$withEnv$1$$anonfun$apply$3.apply(Execution.scala:120)
[error]   org.specs2.execute.ResultExecution$class.execute(ResultExecution.scala:25)
[error]   org.specs2.execute.ResultExecution$.execute(ResultExecution.scala:120)
[error]   org.specs2.execute.Result$$anon$10.asResult(Result.scala:230)
[error]   org.specs2.execute.AsResult$.apply(AsResult.scala:25)
[error]   org.specs2.specification.core.Execution$$anonfun$withEnv$1.apply(Execution.scala:120)
[error]   org.specs2.specification.core.Execution$$anonfun$withEnv$1.apply(Execution.scala:120)
[error]   org.specs2.specification.core.Execution$$anonfun$execute$2$$anonfun$apply$2.apply(Execution.scala:70)
[error]   org.specs2.specification.core.Execution$$anonfun$execute$2$$anonfun$apply$2.apply(Execution.scala:70)
[error]   org.specs2.specification.core.Execution.setResult(Execution.scala:76)
[error]   org.specs2.specification.core.Execution$$anonfun$execute$2.apply(Execution.scala:70)
[error]   org.specs2.specification.core.Execution$$anonfun$execute$2.apply(Execution.scala:70)
[error]   scala.Option.fold(Option.scala:158)
[error]   org.specs2.specification.core.Execution.execute(Execution.scala:70)
[error]   org.specs2.specification.process.DefaultExecutor$$anonfun$executeFragment$1$$anonfun$apply$6.apply(Executor.scala:132)
[error]   org.specs2.specification.process.DefaultExecutor$$anonfun$executeFragment$1$$anonfun$apply$6.apply(Executor.scala:130)
[error]   org.specs2.specification.core.Fragment.updateExecution(Fragment.scala:44)
[error]   org.specs2.specification.process.DefaultExecutor$$anonfun$executeFragment$1.apply(Executor.scala:130)
[error]   org.specs2.specification.process.DefaultExecutor$$anonfun$executeFragment$1.apply(Executor.scala:129)
[error]   org.specs2.specification.process.DefaultExecutor$$anonfun$sequencedExecution$1.executedFragment$lzycompute$1(Executor.scala:104)
[error]   org.specs2.specification.process.DefaultExecutor$$anonfun$sequencedExecution$1.org$specs2$specification$process$DefaultExecutor$class$$anonfun$$executedFragment$1(Executor.scala:104)
[error]   org.specs2.specification.process.DefaultExecutor$$anonfun$sequencedExecution$1$$anonfun$4.apply(Executor.scala:110)
[error]   org.specs2.specification.process.DefaultExecutor$$anonfun$sequencedExecution$1$$anonfun$4.apply(Executor.scala:110)
[error]   scalaz.concurrent.Task$.Try(Task.scala:385)
[error]   org.specs2.data.Processes$$anonfun$start$1.apply(Processes.scala:96)
[error]   org.specs2.data.Processes$$anonfun$start$1.apply(Processes.scala:96)
[error]   scalaz.concurrent.Future$$anonfun$apply$15$$anon$3.call(Future.scala:367)
[error]   scalaz.concurrent.Future$$anonfun$apply$15$$anon$3.call(Future.scala:367)
[error]   java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error]   java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[error]   java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[error]   java.lang.Thread.run(Thread.java:745)
[error]
[error]
[error] This looks like a specs2 exception...
[error] Please report it with the preceding stacktrace at http://github.com/etorreborre/specs2/issues
[error]
[info] ContextTest
[info]
[info] A bar should
[info]   + invoke f on foo
[info]   + invoke g on foo
[info]
[info] Total for specification ContextTest
```

**Comment by nargov** [comment]
Apparently this is caused by a change in specs2, where it no longer catches java.Lang.Error, only Exception. This means that the errors thrown from JMock aren't caught and break the testing framework.

I have opened an issue in the specs2 page [here](https://github.com/etorreborre/specs2/issues/380).

I'll be rolling out a defensive fix tomorrow, regardless.

**Comment by nargov** [comment]
Fixed in version v0.1.16
